src/pybind/mgr/ssh/module.py

```diff
@@ -200,18 +200,23 @@ def config_notify(self):
                     self.get_ceph_option(opt))
             self.log.debug(' native option %s = %s', opt, getattr(self, opt))
 
-    def get_unique_name(self, existing, prefix=None):
+    def get_unique_name(self, existing, prefix=None, forcename=None):
         """
         Generate a unique random service name
         """
         while True:
-            if prefix:
-                name = prefix + '.'
+            if forcename:
+                name = forcename
```

Comment:
:-1: If we have forcename != None, `get_unique_name` is no longer generating anything at all.

Reply:
it's still ensuring the name isn't already taken by another daemon in the cluster

Reply:
Still `forcename` defeats the outer loop entirely. I don't want to be the guy later to refactor this.

Reply:
udpated!